ethylene-core/src/main/java/com/github/steanky/ethylene/core/processor/AsyncFileConfigLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.github.steanky.ethylene.core.processor;
+
+import com.github.steanky.ethylene.core.bridge.ConfigBridge;
+import com.github.steanky.ethylene.core.codec.ConfigCodec;
+import com.github.steanky.ethylene.core.util.FutureUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * Asynchronous specialization of {@link FileConfigLoader}.
+ * @param <TData> the type of data object
+ */
+public class AsyncFileConfigLoader<TData> extends FileConfigLoader<TData> {
+    private final ConfigBridge bridge;
+
+    /**
+     * Constructs a new AsyncFileConfigLoader instance from the given {@link ConfigProcessor}, data object,
+     * {@link Path}, {@link ConfigCodec}, and {@link Executor}.
+     * @param processor the processor used to marshal data
+     * @param defaultData the default data object
+     * @param path the path to read data from and write defaults to
+     * @param codec the {@link ConfigCodec} used to decode the file data
+     * @param executor the executor used to perform read and write operations asynchronously
+     */
+    public AsyncFileConfigLoader(@NotNull ConfigProcessor<TData> processor,
+                                 @NotNull TData defaultData,
+                                 @NotNull Path path,
+                                 @NotNull ConfigCodec codec,
+                                 @NotNull Executor executor) {
+        super(processor, defaultData, path, codec);
+
+        Objects.requireNonNull(executor);
+        this.bridge = new CodecConfigBridge(codec) {
+            @Override
+            protected @NotNull <TReturn> CompletableFuture<TReturn> makeFuture(@NotNull Callable<TReturn> callable) {
+                return FutureUtils.callableToFuture(callable, executor);
+            }
+        };
+    }
+
+    @Override
+    protected @NotNull ConfigBridge getBridge() {
+        return bridge;
+    }
```

Comment:
redundant?

Reply:
nvm, misinterpreted the code - by why not pass in a config bridge instead?

Reply:
defeats the purpose of this class, which is to take some of the burden off of the caller to create these objects

Reply:
why not make it a parameter in the superclass?

Reply:
that would work i think

Reply:
nvm i  have to because i can't reference CodecConfigBridge before supertype constructor is called

Reply:
and i don't want the user to have to specify a ConfigBridge for new instances of FileConfigLoader, that defeats a lot of the purpose

Reply:
why is it even an inner class?

Reply:
not much use outside of AsyncFileConfigLoader + needs access to path field